src/sinks/aws_cloudwatch_logs/request.rs

```diff
@@ -189,9 +200,39 @@ impl Client {
         &self,
         sequence_token: Option<String>,
         mut log_events: Vec<InputLogEvent>,
-    ) -> RusotoFuture<PutLogEventsResponse, PutLogEventsError> {
-        // Sort by timestamp
-        log_events.sort_by_key(|e| e.timestamp);
+    ) -> (
+        RusotoFuture<PutLogEventsResponse, PutLogEventsError>,
+        Option<Vec<InputLogEvent>>,
+    ) {
+        // We will only send events that happened in the 24h window since the oldest event.
```

Comment:
What is the reason for handling this logic here and not doing it all combined in the outer module?

Reply:
At the time, it seemed like it would require the least amount of changes in this place. But, yes, we could handle this by pulling it back to the outer module and passing `Vec<Vec<InputLogEvent>>` to the future.